### PR TITLE
[WIP] Document All The Things

### PR DESCRIPTION
**Pull request by jeffpaul**
- [X] add repo description
- [ ] add setup details
- [ ] add PHPCS server usage details
- [ ] add WPCS usage details
- [ ] add PHP Compatibility usage details
- [ ] add API usage details
- [X] expand Google Lighthouse reporting usage details
- [X] add contributing details
- [X] add table of contents
- [X] pull in content from xwp/wp-tide/readme.md
- [X] add code of conduct
- [ ] update email in code of conduct + contributing
- [X] add issue template
- [x] add pr template

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,45 @@
 
 [![Build Status](https://travis-ci.org/xwp/go-tide.svg?branch=develop)](https://travis-ci.org/xwp/go-tide) [![Coverage Status](https://coveralls.io/repos/github/xwp/go-tide/badge.svg?branch=develop)](https://coveralls.io/github/xwp/go-tide?branch=develop) [![License: MIT](https://img.shields.io/badge/License-MIT-blue.svg)](https://opensource.org/licenses/MIT)
 
+Tide services responsible for polling the WordPress.org APIs for themes and plugins to audit ("Sync Server"), for running audits and sending the results to the Tide API service ("Audit Server"), and for running Google Lighthouse audits against themes in the WordPress.org repository ("Lighthouse Server").
```
Comment (valendesigns) [inline README.md]: Possible changes:
> Tide services are responsible for the following. The Sync Server polls the WordPress.org API's for themes and plugins to process and writes them to a queue. The PHPCS Server reads messages from a queue and runs reports against both plugins and themes, then sends the results back to the Tide API. The Lighthouse Server also reads messages from a queue and runs Google Lighthouse reports against only the themes and send those results back to the Tide API.

File: README.md [hunk 2]
```diff
@@ -2,8 +2,45 @@
...
+
+## Table of Contents
+@todo
+
+### Setup
+@todo
+
+### Usage
+
+#### WPCS Auditing
```
Comment (valendesigns) [inline README.md]: I think we should start to steer away from the term `audit` or `auditing`. Also, I would suggest these title changes:
```
### Usage
@todo

### PHPCS Server
@todo

#### WPCS
@todo

#### PHP Compatibility
@todo

### Lighthouse Server
This section outlines ...
```

Comment (rheinardkorf) [reply README.md]: @valendesigns would you suggest renaming the `audit` package in `wptide/pkg`?  Perhaps `process` ?

File: README.md [hunk 3]
```diff
@@ -2,8 +2,85 @@
...
+This section outlines the components included in the Lighthouse integration with Tide, demonstrates how the integration processes Lighthouse audits of WordPress.org themes, and provides examples of the Tide API with Lighthouse audit results.
+
+#### Status
+Lighthouse auditing of WordPress themes has been integrated by running each theme from wp-themes.com through the Lighthouse CLI, stores the full report, and includes the summary results and a link to the full report in the Tide API. This currently functions locally, XWP is working to get this running on GCP.
+
+#### Components
+The following outlines the components added to Tide in order to integrate Lighthouse in overall Tide auditing of WordPress themes.
+1. Docker Container
+   - Lighthouse CLI
+   - Lighthouse Audit Server binary (lh-server)
```
Comment (valendesigns) [inline README.md]: Correction: `Lighthouse Server binary (lighthouse-server)`

File: README.md [hunk 4]
```diff
@@ -2,8 +2,85 @@
...
+   **Note:** Uses an Alpine Image with working Chromium version. Produces consistent results with Lighthouse Chrome extension. Allows reduced image size of 432MB.
+2. Go Lighthouse Server Source Code
+   - `cmd/lh-server` for binary
```
Comment (valendesigns) [inline README.md]: Correction: `cmd/lighthouse-server` for binary

File: README.md [hunk 5]
```diff
@@ -2,8 +2,85 @@
...
+   - `src` for packages
```
Comment (valendesigns) [inline README.md]: Correction: There is no `src` directory.

File: README.md [hunk 6]
```diff
@@ -2,8 +2,85 @@
...
+   - `vendor` for imported packages
+     - [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go)
```
Comment (valendesigns) [inline README.md]: Correction: We only import two packages.
`github.com/nanobox-io/golang-scribble`
`github.com/wptide/pkg`

File: README.md [hunk 7]
```diff
@@ -2,8 +2,85 @@
...
+     - [github.com/hhatto/gocloc](https://github.com/hhatto/gocloc)
+   - Running Tide cluster
```
Comment (valendesigns) [inline README.md]: Why is this this item here?

File: README.md [hunk 8]
```diff
@@ -2,8 +2,85 @@
...
+
+#### Process
+The following demonstrates how a WordPress theme is run through a Lighthouse audit and has its results stored and returned via the Tide API.
+1. `tide-cluster`
```
Comment (valendesigns) [inline README.md:72]: There is no such thing as a `tide-cluster`

File: README.md [hunk 9]
```diff
@@ -2,8 +2,85 @@
...
+   - Starts all Tide services and listens...
+@todo add screenshot
+2. `lh-server`
```
Comment (valendesigns) [inline README.md]: This is `lighthouse-server`

File: README.md [hunk 10]
```diff
@@ -2,8 +2,85 @@
...
+   - authenticate with Tide API    
+@todo add screenshot
+   - polls a job queue (SQS) for messages to process
+@todo add screenshot
+   - downloads theme and calculates checksum
+@todo add screenshot
+   - runs source through `gocloc` to get code information
+   - scans source for Theme header information (required for Lighthouse Audit)
```
Comment (valendesigns) [inline README.md]: Change `Lighthouse Audit` to `Lighthouse Report`.

File: README.md [hunk 11]
```diff
@@ -2,8 +2,85 @@
...
+@todo add screenshot
+   - runs theme through `lighthouse-cli` at `https://wp-themes.com/<theme-slug>` and keeps polling for next job.
```
Comment (valendesigns) [inline README.md]: The cli command is just called `lighthouse`.

File: README.md [hunk 12]
```diff
@@ -2,8 +2,85 @@
...
+@todo add screenshot
+   - saves full report to a file store (S3)
+@todo add screenshot
```
Comment (valendesigns) [inline README.md]: Missing a step where it takes the full report and generates the detailed report.

File: README.md (hunk 5 above)
Comment (jeffpaul) [reply README.md]: @rheinardkorf is there an update for packages that can be applied to the documentation here?

File: README.md (hunk 7 above)
Comment (jeffpaul) [reply README.md]: @rheinardkorf this is from the source gdoc you wrote. any recollection what this bullet was for?

File: README.md (hunk 8 above)
Comment (jeffpaul) [reply README.md:72]: @rheinardkorf is there an update for the documentation that can be applied here?

File: README.md (hunk 12 above)
Comment (jeffpaul) [reply README.md]: @valendesigns @rheinardkorf are there any commands/screenshots that help highlight that added step?

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+- Try to search for your issue, it may have already been answered or even fixed in the `develop` branch.
+- Check if the issue is reproducible with the latest stable version. If you are using a pre-release, please indicate the specific version you are using.
+- It is **required** that you clearly describe the steps necessary to reproduce the issue you are running into. Issues without clear reproducible steps will be closed immediately.
+- If your issue is resolved but still open, don't hesitate to close it. In case you found a solution by yourself, it could be helpful to explain how you fixed it.
+
+## Pull Request Guidelines
+
+- Checkout a topic branch from `develop` and merge back against `develop`.
+    - If you are not familiar with branching please read [_A successful Git branching model_](http://nvie.com/posts/a-successful-git-branching-model/) before you go any further.
+- Follow the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/coding-standards/).
```
Comment (valendesigns) [inline CONTRIBUTING.md]: This is a Golang codebase, no WordPress code here.

Comment (jeffpaul) [reply CONTRIBUTING.md]: @valendesigns any specific coding standards that are being followed then instead of WPCS?

**Comment by valendesigns** [comment]
I've merged most of these changes into the `api` branch, closing in favor of #42

**Pull request closed without merging**